image/tarexport/load.go

```diff
@@ -77,6 +78,9 @@ func (l *tarexporter) Load(inTar io.ReadCloser, outStream io.Writer, quiet bool)
 		if err != nil {
 			return err
 		}
+		if img.OS != runtime.GOOS {
+			return fmt.Errorf("cannot load %s image on %s", img.OS, runtime.GOOS)
```

Comment:
@thaJeztah I don't see the point of making it a warning. The load will fail hard after the warning is issued.